src/client/routes/Oversikt/Oversikt.jsx

```diff
@@ -1,26 +1,58 @@
-import React, { useEffect } from 'react';
-import { Undertittel } from 'nav-frontend-typografi';
+import React, { useContext, useEffect, useState } from 'react';
+import { Normaltekst, Undertittel } from 'nav-frontend-typografi';
 import { Panel } from 'nav-frontend-paneler';
-import './Oversikt.less';
+import { withRouter } from 'react-router';
 import Lenke from 'nav-frontend-lenker';
-import { withBehandlingContext } from '../../context/BehandlingerContext';
+import PropTypes from 'prop-types';
+import { BehandlingerContext } from '../../context/BehandlingerContext';
+import { InnrapporteringContext } from '../../context/InnrapporteringContext';
+import { extractNameFromEmail } from '../../utils/locale';
 import { oversikttekster } from '../../tekster';
-import { toDate } from '../../utils/date';
-import { withRouter } from 'react-router';
+import { toDate, toDateAndTime } from '../../utils/date';
+import './Oversikt.less';
+
+const Oversikt = ({ history }) => {
+    const behandlingerCtx = useContext(BehandlingerContext);
+    const innrapportering = useContext(InnrapporteringContext);
+    const [behandletSaker, setBehandletSaker] = useState([]);
+    const [ubehandletSaker, setUbehandletSaker] = useState([]);
 
-const Oversikt = withBehandlingContext(
-    ({ behandlinger, fetchAlleBehandlinger, setValgtBehandling, history }) => {
-        useEffect(() => {
-            fetchAlleBehandlinger();
+    const { fetchBehandlingerMedPersoninfo, setValgtBehandling } = behandlingerCtx;
+    const behandlinger = behandlingerCtx.state.behandlinger;
+
+    useEffect(() => {
+        fetchBehandlingerMedPersoninfo();
+    }, []);
+
+    useEffect(() => {
+        let behandlingerUtenFeedback = [];
+        const sakerMedFeedback = behandlinger.reduce((acc, behandling) => {
+            const feedback = innrapportering.feedback.find(f => f.key === behandling.behandlingsId);
+            if (feedback) {
+                const behandletSak = {
+                    søkerName: behandling.personinfo.navn ?? behandling.originalSøknad.aktorId,
+                    submittedDate: feedback.value.submittedDate,
+                    behandlingsId: behandling.behandlingsId,
+                    userName: extractNameFromEmail(feedback.value.userId.email)
+                };
+                acc = [...acc, behandletSak];
+            } else {
```

Comment:
Jeg liker ikke helt en if-else og to forskjellige "outomes" i et reduce-kall.. Jeg tror ikke det er noen krise å dele ting i flere løkker, slik at du kan ha en for å dele inn i behandlinger med og uten feedback og deretter en som mapper behandlede saker til ønsket format.

Partisjonering heter det forresten, å splitte en liste i to ut fra et eller annet predikat.

Reply:
Da tror jeg det blir mest lesbart å kjøre 2 `.filter()` og deretter mappe om `behandledeSaker` til sakerMedFeedback.

Reply:
Da får du to steder hvor predikatet er definert, men go for it :)

Reply:
3 steder :/

Reply:
Går kanskje an å partisjonere listen ved å wrappe reduceren i en `useMemo` og så bruke destructuring assignment for å sette variablene? Da slipper man å ha to `useState`-kall med tilhørende `setState`-funksjoner og man får mer funksjonell kode som ikke har sideeffekter (oppdatere en array utenfor reduceren). Noe som det her kanskje:

```javascript
// Enten ved array destructuring
const [behandledeSaker, ubehandledeSaker] = useMemo(
    () => behandlinger.reduce((acc, behandling) => {
        const feedback = innrapportering.feedback.find(f => f.key === behandling.behandlingsId);
        return feedback
            ? [[...acc[0], toBehandletSak(behandling, feedback)], acc[1]]
            : [acc[0], [...acc[1], behandling]];
    }, [[], []]),
    [innrapportering.feedback, behandlinger]
);

// Eller object destructuring
const { behandledeSaker, ubehandledeSaker } = useMemo(
    () => behandlinger.reduce((acc, behandling) => {
        const feedback = innrapportering.feedback.find(f => f.key === behandling.behandlingsId);
        if (feedback) {
            return {
                ...acc,
                behandledeSaker: [...acc.behandledeSaker, toBehandletSak(behandling, feedback)]
            };
        } else {
            return {
                ...acc,
                ubehandledeSaker: [...acc.ubehandledeSaker, behandling]
            };
        }
    }, { behandledeSaker: [], ubehandledeSaker: [] }),
    [innrapportering.feedback, behandlinger]
);
```
Her har jeg lagt ut noe av logikken i en hjelpefunksjon for at det skal være lettere å følge logikken i reduceren:
```javascript
const toBehandletSak = (behandling, feedback) => ({
    søkerName: behandling.personinfo.navn ?? behandling.originalSøknad.aktorId,
    submittedDate: feedback.value.submittedDate,
    behandlingsId: behandling.behandlingsId,
    userName: extractNameFromEmail(feedback.value.userId.email)
});
```

Reply:
Genialt! 🌟